CalibMuon/DTCalibration/plugins/DTTPAnalyzer.cc

```diff
@@ -66,35 +67,32 @@ class DTTPAnalyzer : public edm::EDAnalyzer {
 #include "TFile.h"
 
 DTTPAnalyzer::DTTPAnalyzer(const edm::ParameterSet& pset)
-    : subtractT0_(pset.getParameter<bool>("subtractT0")), digiLabel_(pset.getParameter<edm::InputTag>("digiLabel")) {
+    : subtractT0_(pset.getParameter<bool>("subtractT0")),
+      digiLabel_(pset.getParameter<edm::InputTag>("digiLabel")),
+      dtGeomToken_(esConsumes()) {
   std::string rootFileName = pset.getUntrackedParameter<std::string>("rootFileName");
   rootFile_ = new TFile(rootFileName.c_str(), "RECREATE");
   rootFile_->cd();
 
   if (subtractT0_)
     tTrigSync_ = DTTTrigSyncFactory::get()->create(pset.getParameter<std::string>("tTrigMode"),
-                                                   pset.getParameter<edm::ParameterSet>("tTrigModeConfig"));
+                                                   pset.getParameter<edm::ParameterSet>("tTrigModeConfig"),
+                                                   consumesCollector());
 }
 
 DTTPAnalyzer::~DTTPAnalyzer() { rootFile_->Close(); }
 
-void DTTPAnalyzer::beginRun(const edm::Run& run, const edm::EventSetup& setup) {
-  // Get the t0 map from the DB
-  if (subtractT0_) {
-    /*ESHandle<DTT0> t0;
-     setup.get<DTT0Rcd>().get(t0);
-     tZeroMap_ = &*t0;*/
-    tTrigSync_->setES(setup);
```

Comment:
I am not sure if this may have a real effect but, why conditions are now retrieved event by event instead of at beginRun?

Reply:
All other places that use that helper retrieve their data on the event. When making the esConsumes call one must state what transition the data will be retrieved from. Having all the uses of the helper retrieve the data on the same transition avoids having to have different esConsumes calls in the helper. Plus, getting data from the EventSetup is really fast.